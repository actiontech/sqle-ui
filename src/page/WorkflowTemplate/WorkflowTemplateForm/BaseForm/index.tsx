import { Button, Form, Input, Select, Space } from 'antd';
import { useForm } from 'antd/lib/form/Form';
import { useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { CreateWorkflowTemplateReqV1AllowSubmitWhenLessAuditLevelEnum } from '../../../../api/common.enum';
import { PageFormLayout } from '../../../../data/common';
import EmitterKey from '../../../../data/EmitterKey';
import useInstance from '../../../../hooks/useInstance';
import EventEmitter from '../../../../utils/EventEmitter';
import { nameRule } from '../../../../utils/FormRule';
import { BaseFormFields, BaseFormProps } from './index.type';
import useStaticStatus from '../../../../hooks/useStaticStatus';
import { useCurrentProjectName } from '../../../ProjectManage/ProjectDetail';

const BaseForm: React.FC<BaseFormProps> = (props) => {
  const { t } = useTranslation();
  const [form] = useForm<BaseFormFields>();
  const { getAuditLevelStatusSelectOption } = useStaticStatus();

  const { updateInstanceList, generateInstanceSelectOption } = useInstance();
  const { projectName } = useCurrentProjectName();
  useEffect(() => {
    updateInstanceList({ project_name: projectName });
  }, [projectName, updateInstanceList]);

  const nextStep = async () => {
    const value = await form.validateFields();
    props.updateBaseInfo(value);
    props.nextStep();
  };

  const resetForm = () => {
    if (!!props.defaultData) {
      form.resetFields([
        'desc',
        'allowSubmitWhenLessAuditLevel',
        'instanceNameList',
      ]);
    } else {
      form.resetFields();
    }
  };

  useEffect(() => {
    const resetAllForm = () => {
      form.resetFields();
    };
    EventEmitter.subscribe(
      EmitterKey.Reset_Workflow_Template_Form,
      resetAllForm
    );
    return () => {
      EventEmitter.unsubscribe(
        EmitterKey.Reset_Workflow_Template_Form,
        resetAllForm
      );
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useEffect(() => {
    if (!!props.defaultData) {
      form.setFieldsValue({
        name: props.defaultData.workflow_template_name,
        desc: props.defaultData.desc,
        allowSubmitWhenLessAuditLevel: props.defaultData
          .allow_submit_when_less_audit_level as
          | CreateWorkflowTemplateReqV1AllowSubmitWhenLessAuditLevelEnum
          | undefined,
        instanceNameList: props.defaultData.instance_name_list,
      });
    } else {
      form.setFieldsValue({
        allowSubmitWhenLessAuditLevel:
          CreateWorkflowTemplateReqV1AllowSubmitWhenLessAuditLevelEnum.warn,
      });
    }
  }, [form, props.defaultData]);

  return (
    <Form {...PageFormLayout} form={form}>
      <Form.Item
        label={t('workflowTemplate.form.label.name')}
        name="name"
        validateFirst={true}
        rules={[
          {
            required: true,
          },
          ...nameRule(),
        ]}
      >
        <Input
          disabled={!!props.defaultData}
          placeholder={t('common.form.placeholder.input', {
            name: t('workflowTemplate.form.label.name'),
          })}
        />
      </Form.Item>
      <Form.Item label={t('workflowTemplate.form.label.desc')} name="desc">
        <Input.TextArea
          placeholder={t('common.form.placeholder.input', {
            name: t('workflowTemplate.form.label.desc'),
          })}
          rows={3}
          className="textarea-no-resize"
        />
      </Form.Item>
      <Form.Item
        label={t('workflowTemplate.form.label.allowSubmitWhenLessAuditLevel')}
        name="allowSubmitWhenLessAuditLevel"
      >
        <Select
          placeholder={t('common.form.placeholder.select', {
            name: t(
              'workflowTemplate.form.label.allowSubmitWhenLessAuditLevel'
            ),
          })}
        >
          {getAuditLevelStatusSelectOption()}
        </Select>
      </Form.Item>
      <Form.Item
        label={t('workflowTemplate.form.label.instanceNameList')}
        name="instanceNameList"
      >
        <Select
          showSearch
          placeholder={t('common.form.placeholder.select', {
            name: t('workflowTemplate.form.label.instanceNameList'),
          })}
          mode="multiple"
        >
          {generateInstanceSelectOption()}
        </Select>
      </Form.Item>
      <Form.Item label=" " colon={false}>
        <Space>
          <Button onClick={resetForm}>{t('common.reset')}</Button>
          <Button onClick={nextStep} type="primary">
            {t('common.nextStep')}
          </Button>
        </Space>
      </Form.Item>
    </Form>
  );
};

export default BaseForm;
