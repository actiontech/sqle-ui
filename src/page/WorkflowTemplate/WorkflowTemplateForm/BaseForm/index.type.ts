import { IWorkflowTemplateDetailResV1 } from '../../../../api/common';
import { WorkflowTemplateDetailResV1AllowSubmitWhenLessAuditLevelEnum } from '../../../../api/common.enum';

export type BaseFormProps = {
  defaultData?: IWorkflowTemplateDetailResV1;
  nextStep: () => void;
  updateBaseInfo: (info: BaseFormFields) => void;
};

export type BaseFormFields = {
  name: string;
  desc?: string;
  instanceNameList?: string[];
  allowSubmitWhenLessAuditLevel?: WorkflowTemplateDetailResV1AllowSubmitWhenLessAuditLevelEnum;
};
