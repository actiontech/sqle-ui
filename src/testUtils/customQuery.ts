import * as domTestingLib from '@testing-library/dom';
import { fireEvent, screen } from '@testing-library/react';

export const getBySelector = (selector: string, baseElement?: Element) => {
  let temp;
  if (baseElement !== undefined) {
    temp = baseElement.querySelectorAll(selector);
  } else {
    temp = document.querySelectorAll(selector);
  }
  if (temp.length === 0) {
    throw domTestingLib.queryHelpers.getElementError(
      `Unable to find an element by: selector=${selector}`,
      document.body
    );
  }
  if (temp.length > 1) {
    throw domTestingLib.queryHelpers.getElementError(
      `Found multiple elements with the selector=${selector}`,
      document.body
    );
  }
  return temp[0];
};

export const getAllBySelector = (selector: string, baseElement?: Element) => {
  let temp;
  if (baseElement !== undefined) {
    temp = baseElement.querySelectorAll(selector);
  } else {
    temp = document.querySelectorAll(selector);
  }
  if (temp.length === 0) {
    throw domTestingLib.queryHelpers.getElementError(
      `Unable to find an element by: selector=${selector}`,
      document.body
    );
  }
  return temp;
};

export const getSelectContentByFormLabel = (label: string) => {
  return getBySelector(
    '.ant-select-selection-item-content',
    screen.getByText(label).parentNode?.parentNode as HTMLDivElement
  );
};

export const getSelectValueByFormLabel = (label: string) => {
  return getBySelector(
    '.ant-select-selection-item',
    screen.getByLabelText(label).parentNode?.parentNode as HTMLDivElement
  );
};

/**
 * 模拟选择 Select 组件中下拉框中的某项数据
 * @param 表单中 Select 组件外部 Form.Item 的label, 注意该 Form.Item 必须有 name
 * @param optionText 需要选中的下拉项
 * @param index 通过文本获取所有的节点会有两个(具体信息可以在网页上查看Select 组件中下拉的节点信息), 正常来说其中第二个是需要来需要用户点击的,
 * 所以这里的 index 默认值为 1. 并且这个节点拥有类名 ant-select-item-option-content, 所以在点击前需要使用 expect 来判断是否为正确的节点.
 */
export const selectOptionByIndex = (
  label: string,
  optionText: string,
  index = 1
) => {
  fireEvent.mouseDown(screen.getByLabelText(label));
  const options = screen.getAllByText(optionText);
  let realIndex = index;
  if (index < 0) {
    realIndex = options.length + index;
  }
  const option = options[realIndex];
  expect(option).toHaveClass('ant-select-item-option-content');
  fireEvent.click(option);
};

/**
 * 具体功能同 selectOptionByIndex, 不过该函数主要针对下拉框选项为一个节点的情况
 * @param label 表单中 Select 组件外部 Form.Item 的label, 注意该 Form.Item 必须有 name
 * @param optionCls 下拉项节点最外层的 class
 * @param index  这里不同于 selectOptionByIndex 的 index, 其代表的就是真实需要选择的下拉项的序号, 从 0 开始.
 */
export const selectCustomOptionByClassName = (
  label: string,
  optionCls: string,
  index: number
) => {
  fireEvent.mouseDown(screen.getByLabelText(label));
  const options = getAllBySelector(`.${optionCls}`);
  let realIndex = index;
  if (index < 0) {
    realIndex = options.length + index;
  }

  const option = options[realIndex];
  expect(option.parentNode).toHaveClass('ant-select-item-option-content');
  fireEvent.click(option);
};

export const getHrefByText = (name: string) => {
  return screen.getByText(name).getAttribute('href');
};

export const getAllHrefByText = (name: string) => {
  return screen.getAllByText(name).map((v) => v.getAttribute('href'));
};
