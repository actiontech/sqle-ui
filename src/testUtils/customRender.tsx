import { render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import {
  BrowserRouter,
  Router,
  RouterProps,
  MemoryRouter,
  MemoryRouterProps,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import { Dictionary } from '../types/common.type';
import { storeFactory } from './mockRedux';
import lightTheme from '../theme/light';
import { ThemeProvider } from '@material-ui/styles';
import { mount, shallow } from 'enzyme';

type RenderParams = Parameters<typeof render>;
type MountParams = Parameters<typeof mount>;
type ShallowParams = Parameters<typeof shallow>;

export const renderWithRouter = (...[ui, option]: [...RenderParams]) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>, option);
};

export const renderWithRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>{ui}</Provider>,
    option
  );
};

export const renderHooksWithRedux = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: Dictionary = {}
) => {
  return renderHook(hooks, {
    wrapper: Provider,
    initialProps: {
      store: storeFactory(storeState),
    },
  } as any);
};

export const renderWithServerRouter = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(<Router {...props}>{ui}</Router>, option);
};

export const renderWithMemoryRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(<MemoryRouter {...props}>{ui}</MemoryRouter>, option);
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(<ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>, option);
};

export const renderWithThemeAndRouter = (
  ...[ui, option]: [...RenderParams]
) => {
  return render(
    <MemoryRouter>
      <ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>
    </MemoryRouter>,
    option
  );
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: lightTheme,
    },
  });
};

export const shallowWithTheme = (...[ui, option]: [...ShallowParams]) => {
  return shallow(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: lightTheme,
    },
  });
};
