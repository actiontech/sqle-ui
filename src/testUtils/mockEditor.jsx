const mockEditor = (props) => {
  // https://github.com/react-monaco-editor/react-monaco-editor/issues/176
  const { editorDidMount, ...otherProps } = props;
  return <input {...otherProps} />;
};

export default mockEditor;
