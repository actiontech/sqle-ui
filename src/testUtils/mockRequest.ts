import { AxiosResponse } from 'axios';
import instance from '../api/instance';
import role from '../api/role';
import ruleTemplate from '../api/rule_template';
import user from '../api/user';
import workflow from '../api/workflow';

export const successData = (data: any, otherData?: any) => {
  return {
    code: 0,
    message: '',
    data,
    ...otherData,
  };
};

export const failedData = (data?: any) => {
  return {
    code: 100,
    message: 'error',
    data,
  };
};

export const resolveImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.resolve({
    status,
    headers,
    config,
    statusText,
    data: successData(data),
  });
};

export const rejectImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.reject({
    status,
    headers,
    config,
    statusText,
    data: failedData(data),
  });
};

export const resolveThreeSecond = (
  data: any,
  {
    status = 200,
    headers = {},
    config = {},
    statusText = '',
    otherData = {},
  } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: successData(data, otherData),
      });
    }, 3000);
  });
};

export const resolveErrorThreeSecond = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: failedData(data),
      });
    }, 3000);
  });
};

export const rejectThreeSecond = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise<AxiosResponse<any>>((_, rej) => {
    setTimeout(() => {
      rej({
        status,
        headers,
        config,
        statusText,
        data: failedData(data),
      });
    }, 3000);
  });
};

export const throwErrorThreeSecond = (
  error = 'error',
  { status = 500, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise((res) => {
    throw new Error(error);
  });
};

export const mockUseInstance = () => {
  const spy = jest.spyOn(instance, 'getInstanceTipListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ instance_name: 'instance1' }])
  );
  return spy;
};

export const mockUseInstanceSchema = () => {
  const spy = jest.spyOn(instance, 'getInstanceSchemasV1');
  spy.mockImplementation(() =>
    resolveThreeSecond({ schema_name_list: ['schema1'] })
  );
  return spy;
};

export const mockUseRole = () => {
  const spy = jest.spyOn(role, 'getRoleTipListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ role_name: 'role_name1' }])
  );
  return spy;
};

export const mockUseRuleTemplate = () => {
  const spy = jest.spyOn(ruleTemplate, 'getRuleTemplateTipsV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ rule_template_name: 'rule_template_name1' }])
  );
  return spy;
};

export const mockUseUsername = () => {
  const spy = jest.spyOn(user, 'getUserTipListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ user_name: 'user_name1' }])
  );
  return spy;
};

export const mockUseWorkflowTemplate = () => {
  const spy = jest.spyOn(workflow, 'getWorkflowTemplateTipsV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ workflow_template_name: 'workflow-template-name-1' }])
  );
  return spy;
};
