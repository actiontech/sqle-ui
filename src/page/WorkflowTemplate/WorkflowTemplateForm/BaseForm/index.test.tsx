import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import BaseForm from '.';
import { mockUseInstance } from '../../../../testUtils/mockRequest';

describe('WorkflowTemplateForm', () => {
  beforeEach(() => {
    jest.useFakeTimers();
    mockUseInstance();
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllTimers();
  });

  test('should match snapshot', () => {
    const { container, rerender } = render(
      <BaseForm nextStep={jest.fn()} updateBaseInfo={jest.fn()} />
    );
    expect(container).toMatchSnapshot();
    rerender(
      <BaseForm
        nextStep={jest.fn()}
        updateBaseInfo={jest.fn()}
        defaultData={{
          workflow_template_name: 'name1',
          desc: 'desc1',
          instance_name_list: ['name1'],
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test('should reset all fields when user click rest button', async () => {
    const { container } = render(
      <BaseForm nextStep={jest.fn()} updateBaseInfo={jest.fn()} />
    );
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    fireEvent.input(screen.getByLabelText('workflowTemplate.form.label.name'), {
      target: { value: 'name1' },
    });
    fireEvent.input(screen.getByLabelText('workflowTemplate.form.label.desc'), {
      target: { value: 'desc' },
    });

    fireEvent.mouseDown(
      screen.getByLabelText('workflowTemplate.form.label.instanceNameList')
    );
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    const instanceOptions = screen.getAllByText('instance1');
    const instance = instanceOptions[1];
    expect(instance).toHaveClass('ant-select-item-option-content');
    fireEvent.click(instance);

    fireEvent.click(screen.getByText('common.reset'));
    expect(container).toMatchSnapshot();
  });

  test('should reset desc and instance name list when user click reset button and prop have defaultData field', async () => {
    render(
      <BaseForm
        nextStep={jest.fn()}
        updateBaseInfo={jest.fn()}
        defaultData={{
          workflow_template_name: 'name1',
          desc: 'desc1',
          instance_name_list: ['instance1'],
        }}
      />
    );
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    fireEvent.click(screen.getByText('common.reset'));
    expect(
      screen.getByLabelText('workflowTemplate.form.label.name')
    ).toHaveAttribute('value', 'name1');
  });

  test('should validate field and call updateBaseInfo when user click next step button', async () => {
    const nextStepFn = jest.fn();
    const updateBaseInfoFn = jest.fn();
    render(
      <BaseForm nextStep={nextStepFn} updateBaseInfo={updateBaseInfoFn} />
    );
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    fireEvent.input(screen.getByLabelText('workflowTemplate.form.label.name'), {
      target: { value: 'name1' },
    });
    fireEvent.input(screen.getByLabelText('workflowTemplate.form.label.desc'), {
      target: { value: 'desc' },
    });

    fireEvent.mouseDown(
      screen.getByLabelText('workflowTemplate.form.label.instanceNameList')
    );
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    const instanceOptions = screen.getAllByText('instance1');
    const instance = instanceOptions[1];
    expect(instance).toHaveClass('ant-select-item-option-content');
    fireEvent.click(instance);
    fireEvent.click(screen.getByText('common.nextStep'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    expect(nextStepFn).toBeCalledTimes(1);
    expect(updateBaseInfoFn).toBeCalledWith({
      name: 'name1',
      desc: 'desc',
      instanceNameList: ['instance1'],
    });
  });
});
