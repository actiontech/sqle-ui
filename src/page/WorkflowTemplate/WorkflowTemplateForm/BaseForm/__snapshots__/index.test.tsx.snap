// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowTemplateForm should match snapshot 1`] = `
<div>
  <form
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-row ant-form-item ant-form-item-has-success"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
      >
        <label
          class=""
          for="allowSubmitWhenLessAuditLevel"
          title="workflowTemplate.form.label.allowSubmitWhenLessAuditLevel"
        >
          workflowTemplate.form.label.allowSubmitWhenLessAuditLevel
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-select ant-select-single ant-select-show-arrow"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="allowSubmitWhenLessAuditLevel_list_0"
                    aria-autocomplete="list"
                    aria-controls="allowSubmitWhenLessAuditLevel_list"
                    aria-haspopup="listbox"
                    aria-owns="allowSubmitWhenLessAuditLevel_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="allowSubmitWhenLessAuditLevel"
                    readonly=""
                    role="combobox"
                    style="opacity: 0;"
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  class="ant-select-selection-item"
                  title="workflowTemplate.auditLevel.warn"
                >
                  workflowTemplate.auditLevel.warn
                </span>
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down ant-select-suffix"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
      >
        <label
          class="ant-form-item-no-colon"
          title=" "
        >
           
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <button
                  class="ant-btn"
                  type="button"
                >
                  <span>
                    common.reset
                  </span>
                </button>
              </div>
              <div
                class="ant-space-item"
              >
                <button
                  class="ant-btn ant-btn-primary"
                  type="button"
                >
                  <span>
                    common.nextStep
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`WorkflowTemplateForm should match snapshot 2`] = `
<div>
  <form
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-row ant-form-item ant-form-item-has-success"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
      >
        <label
          class=""
          for="allowSubmitWhenLessAuditLevel"
          title="workflowTemplate.form.label.allowSubmitWhenLessAuditLevel"
        >
          workflowTemplate.form.label.allowSubmitWhenLessAuditLevel
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-select ant-select-single ant-select-show-arrow"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="allowSubmitWhenLessAuditLevel_list_0"
                    aria-autocomplete="list"
                    aria-controls="allowSubmitWhenLessAuditLevel_list"
                    aria-haspopup="listbox"
                    aria-owns="allowSubmitWhenLessAuditLevel_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="allowSubmitWhenLessAuditLevel"
                    readonly=""
                    role="combobox"
                    style="opacity: 0;"
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  class="ant-select-selection-placeholder"
                >
                  common.form.placeholder.select
                </span>
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down ant-select-suffix"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
      >
        <label
          class="ant-form-item-no-colon"
          title=" "
        >
           
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <button
                  class="ant-btn"
                  type="button"
                >
                  <span>
                    common.reset
                  </span>
                </button>
              </div>
              <div
                class="ant-space-item"
              >
                <button
                  class="ant-btn ant-btn-primary"
                  type="button"
                >
                  <span>
                    common.nextStep
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`WorkflowTemplateForm should reset all fields when user click rest button 1`] = `
<div>
  <form
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
      >
        <label
          class=""
          for="allowSubmitWhenLessAuditLevel"
          title="workflowTemplate.form.label.allowSubmitWhenLessAuditLevel"
        >
          workflowTemplate.form.label.allowSubmitWhenLessAuditLevel
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-select ant-select-single ant-select-show-arrow"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="allowSubmitWhenLessAuditLevel_list_0"
                    aria-autocomplete="list"
                    aria-controls="allowSubmitWhenLessAuditLevel_list"
                    aria-haspopup="listbox"
                    aria-owns="allowSubmitWhenLessAuditLevel_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="allowSubmitWhenLessAuditLevel"
                    readonly=""
                    role="combobox"
                    style="opacity: 0;"
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  class="ant-select-selection-placeholder"
                >
                  common.form.placeholder.select
                </span>
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down ant-select-suffix"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
      >
        <label
          class="ant-form-item-no-colon"
          title=" "
        >
           
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <button
                  ant-click-animating-without-extra-node="false"
                  class="ant-btn"
                  type="button"
                >
                  <span>
                    common.reset
                  </span>
                </button>
              </div>
              <div
                class="ant-space-item"
              >
                <button
                  class="ant-btn ant-btn-primary"
                  type="button"
                >
                  <span>
                    common.nextStep
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`WorkflowTemplateForm should reset audit level when user click reset button and prop have defaultData field 1`] = `
<div>
  <form
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
      >
        <label
          class=""
          for="allowSubmitWhenLessAuditLevel"
          title="workflowTemplate.form.label.allowSubmitWhenLessAuditLevel"
        >
          workflowTemplate.form.label.allowSubmitWhenLessAuditLevel
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-select ant-select-single ant-select-show-arrow"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="allowSubmitWhenLessAuditLevel_list_0"
                    aria-autocomplete="list"
                    aria-controls="allowSubmitWhenLessAuditLevel_list"
                    aria-haspopup="listbox"
                    aria-owns="allowSubmitWhenLessAuditLevel_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="allowSubmitWhenLessAuditLevel"
                    readonly=""
                    role="combobox"
                    style="opacity: 0;"
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  class="ant-select-selection-placeholder"
                >
                  common.form.placeholder.select
                </span>
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down ant-select-suffix"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
      >
        <label
          class="ant-form-item-no-colon"
          title=" "
        >
           
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <button
                  ant-click-animating-without-extra-node="false"
                  class="ant-btn"
                  type="button"
                >
                  <span>
                    common.reset
                  </span>
                </button>
              </div>
              <div
                class="ant-space-item"
              >
                <button
                  class="ant-btn ant-btn-primary"
                  type="button"
                >
                  <span>
                    common.nextStep
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
